x/evm/keeper/statedb.go

```diff
@@ -1,265 +1,589 @@
 package keeper
 
 import (
+	"bytes"
+	"fmt"
 	"math/big"
 
-	"github.com/cosmos/ethermint/x/evm/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	tmtypes "github.com/tendermint/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	ethcmn "github.com/ethereum/go-ethereum/common"
-	ethstate "github.com/ethereum/go-ethereum/core/state"
-	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	ethvm "github.com/ethereum/go-ethereum/core/vm"
+	ethermint "github.com/cosmos/ethermint/types"
+	"github.com/cosmos/ethermint/x/evm/types"
 )
 
+var _ vm.StateDB = &Keeper{}
+
+// csdb defines the internal field values used on the StateDB operations
+type csdb struct {
```

Comment:
Used per transaction per block. Values get reset every block. `TransientState` maybe is a better